_plugins/idp_translations.rb

```diff
@@ -0,0 +1,34 @@
+## This plugin copies the fonts and the images from the identity style guide
+## to the appropriate path on the built site
+##
+## The CSS and JS is already handled by Jekyll and the build:js npm script
+
+require 'fileutils'
+require 'uri'
+require 'net/http'
+require 'json'
+require 'open-uri'
+
+DOMAIN = 'https://secure.login.gov'
+
+def copy_translations(origin_uris, destination)
+  origin_uris.each do |uri|
+    filename = uri.gsub(DOMAIN, '').gsub(%r{^/}, '')
+    dest_filename = File.join(destination, filename)
+    next if File.exist?(dest_filename)
+
+    download = URI.open("#{DOMAIN}#{uri}")
+    FileUtils.mkdir_p(File.dirname("#{destination}#{uri}"))
+    IO.copy_stream(download, "#{destination}#{uri}")
```

Comment:
We should create the directory before performing the file download operations. It's best to keep any operations related to the request as short as possible.

The call to `URI.open` will [read the entire file into an in-memory buffer](https://github.com/ruby/open-uri/blob/master/lib/open-uri.rb#L347), contrary to what some guides indicate. If you want to avoid memory overhead, then I suggest creating [`Net::HTTP::Get`](https://ruby-doc.org/stdlib-2.7.0/libdoc/net/http/rdoc/Net/HTTP/Get.html) and using [`Net::HTTPResponse.open`](https://ruby-doc.org/stdlib-2.7.0/libdoc/net/http/rdoc/Net/HTTPResponse.html#method-i-read_body) directly to stream into the file or checking if something like [HTTPX](https://gitlab.com/os85/httpx) might work.

```suggestion
    FileUtils.mkdir_p(File.dirname("#{destination}#{uri}"))
    download = URI.open("#{DOMAIN}#{uri}")
    IO.copy_stream(download, "#{destination}#{uri}")
```